Spigot-API-Patches/0132-Custom-plugin-advancement-API.patch

```diff
@@ -0,0 +1,260 @@
+From 1791bc0bf4c38b3d838b09c2d3eaa998a1086e86 Mon Sep 17 00:00:00 2001
+From: miclebrick <miclebrick@outlook.com>
+Date: Sat, 1 Sep 2018 12:28:16 -0400
+Subject: [PATCH] Custom plugin advancement API
+
+
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 2912bdae..36795d33 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -13,6 +13,7 @@ import java.util.UUID;
+ import java.util.logging.Logger;
+ 
+ import org.bukkit.Warning.WarningState;
++import org.bukkit.advancement.AdvancementManager;
+ import org.bukkit.boss.BarColor;
+ import org.bukkit.boss.BarFlag;
+ import org.bukkit.boss.BarStyle;
+@@ -863,6 +864,16 @@ public interface Server extends PluginMessageRecipient {
+      */
+     ScoreboardManager getScoreboardManager();
+ 
++    // Paper start - custom advancement API
++
++    /**
++     * Gets the instance of the advancement manager.
++     *
++     * @return the advancement manager
++     */
++    AdvancementManager getAdvancementManager();
++    // Paper end
++
+     /**
+      * Gets an instance of the server's default server-icon.
+      *
+@@ -922,12 +933,12 @@ public interface Server extends PluginMessageRecipient {
+ 
+     /**
+      * Create a ChunkData for use in a generator.
+-     * 
++     *
+      * See {@link ChunkGenerator#generateChunkData(org.bukkit.World, java.util.Random, int, int, org.bukkit.generator.ChunkGenerator.BiomeGrid)}
+-     * 
++     *
+      * @param world the world to create the ChunkData for
+      * @return a new ChunkData for the world
+-     * 
```

Comment:
Wut?

Reply:
IDK

Reply:
Set your IDE to only trim trailing whitespace on VCS changed lines.